lib/matplotlib/legend.py

```diff
@@ -584,10 +552,6 @@ def __init__(self, parent, handles, labels,
         else:
             self.get_frame().set_alpha(framealpha)
 
-        tmp = self._loc_used_default
-        self._set_loc(loc)
-        self._loc_used_default = tmp  # ignore changes done by _set_loc
```

Comment:
Those lines somehow got lost during rebase. I have no idea what they are used for. Anyone else maybe?

Reply:
ping @timhoffm about this code...

Reply:
This must stay in, and probably get a more extensive comment.

It's a hack to remember if we're using `loc='best'` as default or if it was explicitly set by the user. It's used to determine whether to issue a warning with `loc='best'` and many plot elements. The warning should only be issued with for the default, not when the user explicitly sets 'best'.

See https://github.com/matplotlib/matplotlib/pull/12455#pullrequestreview-176063630
and the link therein: https://github.com/timhoffm/matplotlib/commit/917b5205645af78ea2a419293d147e97cce21422